.github/workflows/int-and-connected-test-run.yml

```diff
@@ -30,8 +36,46 @@ on:
   schedule:
     - cron: '0 4 * * *' # run at 4 AM UTC, 12PM EST.
 jobs:
+  test_check:
+    name: "Checkout, verify and run unit tests"
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+    if: "!contains( github.event.pull_request.labels.*.name, 'skip-ci')"
+    runs-on: ${{ matrix.os }}
+    ## skips if pr label is 'skip-ci'
+    # run a local Postgres container in Docker for the basic check tests
+    services:
+      postgres:
+        image: postgres:11
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@v2
+      - name: "Cache build"
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CACHE_PATHS }}
+          key: ${{ runner.os }}-build-unit }}
```

Comment:
Instead of using the `cache` action with a manually specified key, you may just want to use the `setup-java` action in a build step and use the built in gradle cache. It will persist the build cache between steps, job, and runs: https://github.com/DataBiosphere/terra-data-catalog/blob/main/.github/workflows/build-and-test.yml#L50

We previously used the cache action with a manual key, but this would often leave gradle in a confused state and sometimes broke the CI build.

Reply:
I'm not sure it'll work since the setup-java action doesn't cache the directories I want (and it's just using github caches under the covers)